packages/clay-css/src/scss/components/_buttons.scss

```diff
@@ -1,261 +1,147 @@
 .btn {
-	background-color: transparent;
-	border: $btn-border-width solid transparent;
-
-	@include border-radius($btn-border-radius, 0);
-	@include box-shadow($btn-box-shadow);
-
-	color: $body-color;
-	cursor: $btn-cursor;
-	display: inline-block;
-	font-family: $btn-font-family;
-	font-size: $btn-font-size;
-	font-weight: $btn-font-weight;
-	line-height: $btn-line-height;
-	padding: $btn-padding-y $btn-padding-x;
-	text-align: center;
-	text-transform: none;
+	background-color: var(--btn-bg, transparent);
+	border: var(--btn-border-width, 1px) var(--btn-border-style, solid)
+		var(--btn-border-color, transparent);
+	border-radius: var(--btn-border-radius, 0.25rem);
+	box-shadow: var(--btn-box-shadow);
+	color: var(--btn-color, inherit);
+	cursor: var(--btn-cursor, pointer);
+	display: var(--btn-display, inline-grid);
+	font-family: var(--btn-font-family);
+	font-size: var(--btn-font-size);
+	font-weight: var(--btn-font-weight);
+	line-height: var(--btn-line-height);
+	padding: var(--btn-padding-y, 0.375rem) var(--btn-padding-x, 0.75rem);
+	place-content: var(--btn-place-content, center);
+	opacity: var(--btn-opacity);
 
 	@include transition($btn-transition);
 
-	user-select: none;
-	vertical-align: middle;
-	white-space: $btn-white-space;
-
-	@include clay-scale-component {
-		font-size: $btn-font-size-mobile;
-		padding-bottom: $btn-padding-y-mobile;
-		padding-left: $btn-padding-x-mobile;
-		padding-right: $btn-padding-x-mobile;
-		padding-top: $btn-padding-y-mobile;
-	}
-
 	&:hover {
-		color: $body-color;
-		text-decoration: none;
+		--btn-color: inherit;
+
+		@at-root :where(a#{&}) {
```

Comment:
This is just an example.

- `text-decoration: none;` is needed for `<a>` elements
- with `:where` we are creating a `a.btn { ... }` rule without specificity
- `:where` is not fully supported yet